File: src/main/java/edu/uchicago/gerber/_08final/mvc/model/ShieldFloater.java
package edu.uchicago.gerber._08final.mvc.model;

import edu.uchicago.gerber._08final.mvc.controller.CommandCenter;
import edu.uchicago.gerber._08final.mvc.controller.Game;
import edu.uchicago.gerber._08final.mvc.controller.Sound;

import java.awt.*;
import java.util.LinkedList;

public class ShieldFloater extends Floater {
	//spawn every 25 seconds
	public static final int SPAWN_SHIELD_FLOATER = Game.FRAMES_PER_SECOND * 25;
	public ShieldFloater() {
		setColor(Color.CYAN);
		setExpiry(260);
	}

	@Override
	public void remove(LinkedList<Movable> list) {
		super.remove(list);
		//if getExpiry() > 0, then this remove was the result of a collision, rather than natural mortality
		if (getExpiry() > 0) {
			Sound.playSound("shieldup.wav");
		    CommandCenter.getInstance().getFalcon().setShield(Falcon.MAX_SHIELD);
	   }

	}


}

File: src/main/java/edu/uchicago/gerber/_08final/mvc/model/ShieldFloater.kt
package edu.uchicago.gerber._08final.mvc.model

import edu.uchicago.gerber._08final.mvc.controller.Game
import java.awt.Color

class ShieldFloater : Floater() {

    companion object {
        //spawn every 25 seconds
        const val SPAWN_SHIELD_FLOATER = Game.FRAMES_PER_SECOND * 25
    }
    init {
        color = Color.CYAN
        expiry = 260
    }


}
